pkg/mcp/client/client.go

```diff
@@ -263,6 +280,8 @@ func (c *Client) Run(ctx context.Context) {
 			log.Info("(re)trying to establish new MCP stream")
 			var err error
 			if c.stream, err = c.client.StreamAggregatedResources(ctx); err == nil {
+				// This technically counts the initial connection as well.
+				c.reporter.RecordReconnect()
```

Comment:
Maybe rename to StreamCreateSuccess() to handle initial plus subsequent reconnects?